### DISCUSSION TITLE
Binding to enums

### DISCUSSION THREAD

**Discussion by inertia666** [Blazored/Modal#333] [Q&A] [answered]
Is there a simple way or workaround solution to binding enums in an object?

Here is a partial from my model

```
    public class FanState
    {
        /// <summary>
        /// Is the fan powered on?
        /// </summary>
        public TimestampedValue<PowerState> Power { get; set; } = new TimestampedValue<PowerState>();
```

And here I am trying to bind this in the modal, which doesn't work.

I'd like to use an InputSelector so the user can select the text and the enum value is used instead. Similar to the code below but with a selector of course.

```
    <div class="form-group">
        <label for="state-power">Power</label>
        <input @bind="@Fan.State.Power.Value" type="text" class="form-control" id="state-power" />
    </div>
```

At the moment I am playing around trying to understand things and have this, which of course is bound.

```
<div class="form-group">
        <label for="state-power">Power</label>

        <select>
            @foreach (var value in Enum.GetValues(typeof(PowerState)))
            {
                <option>@value</option>
            }
        </select>
 </div>
```

**Comment by larsk2009** [comment, marked as answer]
I don't think you can bind enums to an `InputSelect`, or well not automatically that is. You can do something [like this](https://www.pragimtech.com/blog/blazor/bind-input-select-element-to-enum-in-blazor/) possibly.

**Comment by inertia666** [comment]
Just to update...the problem wasn't so much the Enum but the object type. I had to write a callback function and create the object type there.
